pkg/build/builder/docker_test.go

```diff
@@ -133,14 +136,206 @@ RUN echo "hello world"
 			t.Errorf("test[%d]: %v", i, err)
 			continue
 		}
-		replaceLastFrom(got, test.image)
+		replaceLastFrom(got, test.image, "")
 		if !bytes.Equal(dockerfile.Write(got), dockerfile.Write(want)) {
 			t.Errorf("test[%d]: replaceLastFrom(node, %+v) = %+v; want %+v", i, test.image, got, want)
 			t.Logf("resulting Dockerfile:\n%s", dockerfile.Write(got))
 		}
 	}
 }
 
+func TestAppendPostCommit(t *testing.T) {
+	type want struct {
+		Err bool
+		Out string
+	}
+	tests := []struct {
+		description string
+		original    string
+		postCommit  buildapiv1.BuildPostCommitSpec
+		from        *corev1.ObjectReference
+		build       []buildapiv1.ImageSource
+		want        want
+	}{
+		{
+			description: "basic multi-part bash command",
+			original: heredoc.Doc(`
+				FROM busybox
+				RUN echo "hello world"
+				`),
+			postCommit: buildapiv1.BuildPostCommitSpec{
+				Command: []string{"echo", "hello"},
+			},
+			want: want{
+				Out: heredoc.Doc(`
+				FROM busybox as appimage
+				RUN echo "hello world"
+				FROM appimage
+				RUN echo hello
+				FROM appimage
+				`),
+			},
+		},
+		{
+			description: "basic bash command with args",
+			original: heredoc.Doc(`
+				FROM busybox
+				RUN touch /tmp/hello
+				`),
+			postCommit: buildapiv1.BuildPostCommitSpec{
+				Command: []string{"ls"},
+				Args:    []string{"-l", "/tmp/hello"}},
+			want: want{
+				Out: heredoc.Doc(`
+				FROM busybox as appimage
+				RUN touch /tmp/hello
+				FROM appimage
+				RUN ls -l /tmp/hello
+				FROM appimage
+				`),
+			},
+		},
+		{
+			description: "basic bash script",
+			original: heredoc.Doc(`
+				FROM busybox
+				RUN echo "hello world"
+				`),
+			postCommit: buildapiv1.BuildPostCommitSpec{
+				Script: "echo hello $1 world",
+			},
+			want: want{
+				Out: heredoc.Doc(`
+				FROM busybox as appimage
+				RUN echo "hello world"
+				FROM appimage
+				RUN /bin/sh -ic 'echo hello $1 world'
+				FROM appimage
+				`),
+			},
+		},
+		{
+			description: "basic bash script with args",
+			original: heredoc.Doc(`
+				FROM busybox
+				RUN echo "hello world"
+				`),
+			postCommit: buildapiv1.BuildPostCommitSpec{
+				Script: "echo",
+				Args:   []string{"hello", "$1", "world"},
+			},
+			want: want{
+				Out: heredoc.Doc(`
+				FROM busybox as appimage
+				RUN echo "hello world"
+				FROM appimage
+				RUN /bin/sh -ic 'echo hello $1 world'
+				FROM appimage
+				`),
+			},
+		},
+		{
+			description: "multi-stage basic multi-part bash command",
+			original: heredoc.Doc(`
+				FROM busybox as appimage
+				RUN echo "hello world"
+				FROM appimage
+				RUN touch /tmp/hello
+				`),
+			postCommit: buildapiv1.BuildPostCommitSpec{
+				Command: []string{"echo", "hello", "$1", "world"},
+			},
+			want: want{
+				Out: heredoc.Doc(`
+				FROM busybox as appimage
+				RUN echo "hello world"
+				FROM appimage as appimage
```

Comment:
does this actually work in docker, since you've declared the same alias twice?  does the second one override the first one?

Reply:
@bparees 
Nope, it gives an error.  I see two options here:
 - Come up with a more unique name for the alias that gets added.
 - Re-implement the uuid stuff that I had before.
I am actually leaning towards number one, as I am sure we can come up with a static name that is unique enough for 99% of the time.  The uuid also requires more code, and more code to test.

Thoughts?

Reply:
pick something sufficiently unique, users who manage to collide can update their dockerfile to change their alias to something else.

Reply:
@bparees updates in latest push, ptal